src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs

```diff
@@ -314,8 +314,6 @@ public async Task TestResponseWithoutContentEvents()
                 Assert.Equal("System.Net.Http.Desktop.HttpRequestOut.Ex.Stop", stopEvent.Key);
                 HttpWebRequest stopRequest = ReadPublicProperty<HttpWebRequest>(stopEvent.Value, "Request");
                 Assert.Equal(startRequest, stopRequest);
-                HttpStatusCode status = ReadPublicProperty<HttpStatusCode>(stopEvent.Value, "StatusCode");
-                Assert.NotNull(status);
```

Comment:
Is deleting this correct?  Or should it be validating that HttpStatusCode is a specific value? cc: @lmolkova

Reply:
no this is not correct, this checks that there is a payload with name "StatusCode". It is possible to check status value instead - I guess it's 200 (but this is not the point of the test).

Reply:
> this checks that there is a payload with name "StatusCode"

@lmolkova, it's not, though.  HttpStatusCode is an enum; it can never be null.  That's why this needs to either be deleted or changed.

Reply:
> no this is not correct, this checks that there is a payload with name "StatusCode". It is possible to check status value instead - I guess it's 200 (but this is not the point of the test).

That was what prompted me to remove it. I felt the point of the test was accomplished in the check between started and stopped. I'm happy to validate that 200 is right it and add it back. Could/should the status check be a test of its own given your comment?

Reply:
Well, the point is to validate that anonymous DiagnosticSource payload had StatusCode property.   

Here it is added:
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs#L675

Let's say code there will not include StatusCode as a property then this code in the test will throw (`property` will be null):

```csharp
        private static T ReadPublicProperty<T>(object obj, string propertyName)
        {
            Type type = obj.GetType();
            PropertyInfo property = type.GetProperty(propertyName, BindingFlags.Instance | BindingFlags.Public);
            return (T)property.GetValue(obj);
        }
```

So the test will fail, and maybe it's ok to leave it like that.

Reply:
Thanks for the insights @lmolkova, I'm updating and validating this now.